tests/test-dirs/syntax-document/language-extensions.t/run.t

```diff
@@ -0,0 +1,247 @@
+  $ syn_doc 2:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Variant Type"
+on type a3..
+  $ syn_doc 3:6 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Empty Variant type"
+on type a4..
+  $ syn_doc 4:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value.name'
+  "Record type"
+on type a5..
+  $ syn_doc 5:5 \
+  > -filename ./types.ml < ./types.ml | jq '.value'
+  "No documentation found"
+
+// Private types
+// Extensible
+  $ cat > p-types.ml << EOF
+  > type b1 = private ..
+  > type b2 = private A
+  > type b3 = private A of int
+  > type b4 = private { x:int }
+  > type b5 = private int 
+  > module N : sig
+  >   type t = private int
+  >   val of_int: int -> t
+  >   val to_int: t -> int
+  > end = struct
+  >   type t = int
+  >   let of_int n = assert (n >= 0); n
+  >   let to_int n = n
+  > end
+  > EOF
+  on type b1..
+  $ syn_doc 1:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Extensible Variant Type"
+on type b2..
+  $ syn_doc 2:14 \
+  > -filename ./private-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type"
+on type b3..
+  $ syn_doc 3:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type"
+on type b4..
+  $ syn_doc 4:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Record Type"
+on b5..
+  $ syn_doc 5:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type Abbreviation"
+on type t = private int..
+  $ syn_doc 7:14 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value.name'
+  "Private Type Abbreviation"
+on type t = int..
+  $ syn_doc 11:7 \
+  > -filename ./p-types.ml < ./p-types.ml | jq '.value'
+  "No documentation found"
+
+
+// Locally abstract data types
+  $ cat > locally-abstract-dt.ml << EOF 
+  > let f = fun (type t) (x: t) -> x = x
+  > let sort_uniq (type s) (cmp : s -> s -> int) =
+  >   let module S = Set.Make(struct type t = s let compare = cmp end) in
+  >   fun l ->
+  >     S.elements (List.fold_right S.add l S.empty)
+  > EOF
+// Locally abstract data types
+on type t..
+  $ syn_doc 1:17 \
+  > -filename ./locally-abstract-dt.ml < ./locally-abstract-dt.ml | jq '.value.name'
+  "Locally Abstract Type"
+
+// First class Modules
+  $ cat > first-class-modules.ml << EOF
+  > type picture = { x : int; y : int }
+  > module type DEVICE = sig
+  >   val draw : picture -> unit
+  > end
+  > let devices : (string, (module DEVICE)) Hashtbl.t = Hashtbl.create 17
+  > module SVG = struct end
+  > module PNG = struct end
+  > let _svg = Hashtbl.add devices "SVG" (module SVG : DEVICE)
+  > let _png = Hashtbl.add devices "PNG" (module PNG : SVG)
+  > EOF
+on type picture
+  $ syn_doc 1:6 \
+  > -filename ./first-class-modules.ml < ./first-class-modules.ml | jq '.value.name'
+  "Record type"
+on (module SVG : DEVICE)
+  $ syn_doc 6:43 \
+  > -filename ./first-class-modules.ml < ./first-class-modules.ml | jq '.value.name'
+  "First class module"
```

Comment:
This is not testing what the comment says: 
`(module SVG : DEVICE)` is on line 8, not 6.

And it also shows that there is an issue: on line 6, this is not a first class module but a normal module. The output should be "no documentation".

And if I test it on line 8 it doesn't find a first class module, so there is definitely an issue here.

We should keep the two tests: on on line 6 and on on line 8 and fix the behavior.

(same remark applies to the next test)